### DISCUSSION TITLE
PlugData v0.4

### DISCUSSION THREAD

**Discussion by timothyschoen** [plugdata-team/plugdata#40] [General]
New version with many new features and bugfixes. Should be much more stable and usable than previous versions!

Thanks to @60-hz,  @alfonso73 and @FlachyJoe for bug reports and feature requests

Changelog:
- Better graph on parent management
- Change keyboard shortcuts in settings
- Improved search path dialog
- Better handles for resizing objects
- Large improvements to inspector and console
- Presentation mode
- MIDI blinker
- Better level meter
- Pin sidebar to stop from automatically changing
- Automatically shows console when editing values
- Nicer look for toggle, bang and numbox
- Dotted background to indicate edit/lock mode
- More flexible connection cord manipulation (might still need more work)
- Curved connection cord replaces straight cord mode
- Supports up to 16 stereo audio channels in DAW (formerly 2)
- Supports up to 512 automatable parameters (formerly 16)
- Fixed send/receive bugs
- Fixed object position and size bugs
- Improved thread safety
- Drastically reduced cpu use
- Improved undo/redo consistency
- Auto-scroll when dragging close to edges, middle-mouse scroll
- Zoom by ctrl+scroll or pinch on trackpads
- Fixed gaps when dragging over array component
- Fixed many other bugs
- Cleaned up a lot of code
- MacOS build is now notarised

Known issues:
- Problems with audio on macOS standalone
- Some pd patches look messy in PlugData (#36, #22)
- Loading C externals fails (#34)

<hr /><em>This discussion was created from the release <a href='https://github.com/timothyschoen/PlugData/releases/tag/v0.4'>PlugData v0.4</a>.</em>

**Comment by alfonso73** [comment]
oops little bug....no "add object" (the plus button is greyed out) when in editing mode

**Reply by timothyschoen** [reply]
I noticed this, but it's only this way on Windows, and it's a total mystery to me how this can happen... I might replace the windows versions with one where I just manually flip it, but that's obviously not ideal...

**Reply by timothyschoen** [reply]
I've uploaded a new version where I flipped it but it turns out it's not really flipped, now the behaviour is just weird. I will fix this very soon, and update the Windows binaries...

I've had more problems with using juce::Value on Windows, I hope it's not causing any other bugs...

**Reply by timothyschoen** [reply]
should be fixed now!
